### PR TITLE
Fixing the inheritance logic

### PR DESCRIPTION
**Pull request by sarangan12**
A swagger file can have 'Resource'/'SubResource' models defined in them. 

In the current design, these two models have defined structure in ms_rest_azure. When the provided structure matches with the one defined then the defined ones are used. Else, a new class is defined per gem and this new class derives from the standard ones in MsRestAzure. 

But, the above approach does have a problem. When the 'Resource' in swagger has properties such as 'name', 'p1', 'p2', 'p3', the above approach fails. 

In order to handle this condition, a new design is provided - 'Generate Resource/SubResource per gem'. 

**What are the changes included in this PR?**
1. Any new code that was added originally to fix the inheritance logic has been removed. 
2. There is no more logic around x-ms-azure-resource & x-ms-azure-external extensions. I have discussed with Amar too and confirmed that this does not have any impact. 
3. AzureModelTemplate.cshtml was introduced originally to handle this issue. Since the logic has been changed, there is no reason to keep it. So, I removed. 

**How did I test?**
1. I opened all the models in all the services and checked manually. 
2. I have written a script to load all the models in all the services and confirmed there are no issues

@vishrutshah @veronicagg @salameer Please review and approve,

### CODE REVIEW

File: src/azure/Model/CompositeTypeRba.cs [hunk 1]
```diff
@@ -48,69 +48,13 @@ public override string GetBaseTypeName()
             {
                 string typeName = this.BaseModelType.Name;
```
Comment (veronicagg) [inline src/azure/Model/CompositeTypeRba.cs]: should we not declare this variable now, and just return   " < " + this.BaseModelType.Name; ? else return string.Empty?

**Comment by veronicagg** [comment]
and thanks for the extensive testing on this!

File: src/azure/Model/CompositeTypeRba.cs (hunk 1 above)
Comment (sarangan12) [reply src/azure/Model/CompositeTypeRba.cs]: @veronicagg Yes. the 'typeName' is no longer required. I have removed with the latest commit.

**Comment by sarangan12** [comment]
@vishrutshah @veronicagg I have added the testcases around the inheritance logic.

**Comment by sarangan12** [comment]
@veronicagg Just to be on the safe side, I have created issue https://github.com/Azure/azure-sdk-for-ruby/issues/1010 and assigned it to next-arm and myself. I will just wait for one more release before taking that step :-)

File: test/azure/RspecTests/resource_inheritance_spec.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    # Should generate Resource class
+    resource = modules.const_get('Resource')
+    expect(resource.is_a?(Class)).to be_truthy
+  end
+
+  it 'should generate models with expected inheritance' do
+    modules = Module.const_get('AzureResourceInheritanceModule::Models')
+
+    # Should generate InheritMsRestAzureResource with super class as MsRestAzure::Resource
```
Comment (veronicagg) [inline test/azure/RspecTests/resource_inheritance_spec.rb]: I believe this comment is out of date

File: test/azure/RspecTests/resource_inheritance_spec.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
+# encoding: utf-8
+
+$: << 'RspecTests/Generated/azure_resource_inheritance'
+
+require 'rspec'
+require 'generated/resource_inheritance'
+
+include AzureResourceInheritanceModule
+
+describe 'ResourceInheritance' do
```
Comment (veronicagg) [inline test/azure/RspecTests/resource_inheritance_spec.rb]: should this be renamed, as it's not testing inheritance for resource anymore? but validating that all models are generated for any inheritance relationship?

**Comment by vishrutshah** [comment]
Just adding Issue link here..

Fixing https://github.com/Azure/azure-sdk-for-ruby/issues/957

File: test/azure/RspecTests/resource_inheritance_spec.rb (hunk 2 above)
Comment (sarangan12) [reply test/azure/RspecTests/resource_inheritance_spec.rb]: Yes. My bad... Fixed it

**Comment by olydis** [comment]
# publish job

**Pull request merged**